Automattic/wp-calypso
# wp-admin Site Default: Link directly to wp-admin

Closes https://github.com/Automattic/dotcom-forge/issues/4274

## Proposed Changes

This PR changes the default links on /sites page from the site thumbnail and site title (both on the grid and table view) to go to `wp-admin` instead of `Calypso` when the `wp-admin` is selected as the default interface. 

## Testing Instructions

1. Pull changes from this branch or navigate to Calypso live link
2. Create an Atomic site
3. Navigate to `https://wordpress.com/hosting-config/{yourAtomicSite}`
4. Make sure you enabled Hosting Features
5. Scroll to the `Switch to wp-admin` card 
6. Toggle on the `wp-admin` interface like so:

<img width="958" alt="Screenshot 2023-10-20 at 2 14 14 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/f1e7e63d-cab0-45f5-ac93-b678e39c8c52">

7. Navigate to `/sites` page
8. In the table view, hover over the thumbnail and the site title of your Atomic site
9. Confirm that the link shows as `wp-admin`

<img width="1067" alt="Screenshot 2023-10-20 at 2 17 56 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/42c8f66d-8cfd-406b-a558-bcbf646ac278">

10. Click on thumbnail or / and on site title and confirm you are being brought to `wp-admin`
11. Go back to  `/sites` page
12. Switch to the grid view
13. Hover over the site title or site thumbnail and confirm that the link shows as `wp-admin`
14. Click on the link 
15. Confirm you are being brought to `wp-admin`

**Sidenotes:**

* I added an extra check for `hostingFeatures` condition along with the check for the interface option to be set to `wp-admin`. Technically, it is not possible to set this option anywhere but on the Atomic site but it is still there (meaning the option) on Simple and JP sites, so I prefer to have a check to be extra sure;
* I manually added `isWpAdminDefaultSite ` in the grid and table view. I am wondering if we might want to create a selector going forward if we are planning to default more places in Calypso to `wp-admin`

## Pre-merge Checklist
